src/Features/Core/Portable/Completion/LSPCompletionProvider.cs

```diff
@@ -0,0 +1,17 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Collections.Immutable;
+
+namespace Microsoft.CodeAnalysis.Completion
+{
+    internal abstract class LSPCompletionProvider : CommonCompletionProvider
+    {
+        /// <summary>
+        /// Defines the set of possible non-identifier trigger characters for this completion provider.
+        /// Used by the LSP server to determine the trigger character set for completion.
+        /// </summary>
+        internal virtual ImmutableHashSet<char> TriggerCharacters { get; } = ImmutableHashSet<char>.Empty;
```

Comment:
```suggestion
        public virtual ImmutableHashSet<char> TriggerCharacters { get; } = ImmutableHashSet<char>.Empty;
```

Reply:
Wait.  if this is now just for us, then have this be abstract, so all Roslyn subclasses are forced to implement this.